scandium-core/src/main/java/org/eclipse/californium/scandium/dtls/ServerHandshaker.java

```diff
@@ -782,41 +820,50 @@ private void addServerHelloExtensions(final CipherSuite negotiatedCipherSuite, f
 			if (clientHello.getSupportedPointFormatsExtension() != null) {
 				// if we chose a ECC cipher suite, the server should send the
 				// supported point formats extension in its ServerHello
-				List<ECPointFormat> formats = Arrays.asList(ECPointFormat.UNCOMPRESSED);
-				extensions.addExtension(new SupportedPointFormatsExtension(formats));
+				extensions.addExtension(SupportedPointFormatsExtension.DEFAULT_POINT_FORMATS_EXTENSION);
 			}
 		}
 	}
 
 	/**
 	 * Determines the elliptic curve to use during the EC based DH key exchange.
 	 * 
-	 * @param clientHello
-	 *            the peer's <em>CLIENT_HELLO</em> message containing its
-	 *            preferred elliptic curves
-	 * @return the selected curve or {@code null} if server and client have no curves in common
+	 * @param clientHello the peer's <em>CLIENT_HELLO</em> message containing
+	 *            its preferred elliptic curves
+	 * @return a list of common supported curves. Maybe empty, if server and
+	 *         client have no curves in common
+	 * 
```

Comment:
it seems to me that javadoc is not correct ? 
We don't search the EC but all the supported group handled by both side ordered by client preferences ?

Reply:
Yes, but in order to use them for ECDHE (and sometimes also to check the public key or certificate chain, depending on other parameters).